bin/runtime/src/lib.rs

```diff
@@ -681,7 +681,7 @@ impl pallet_contracts::Config for Runtime {
     type DeletionQueueDepth = DeletionQueueDepth;
     type DeletionWeightLimit = DeletionWeightLimit;
     type Schedule = Schedule;
-    type CallStack = [pallet_contracts::Frame<Self>; 31];
+    type CallStack = [pallet_contracts::Frame<Self>; 16];
```

Comment:
Not 15? There's one more implicit frame for the initial contract call

Reply:
Our depth is 17 and call stack is 16 :)

Reply:
What @pmikolajczyk41 meant, I think, is that we want to allow for 16 in total, including the starting frame, and at least the `integrity_test` will add 1 to the `CallStack::size` return value https://github.com/paritytech/substrate/blob/8d3eace503c3956fbeda321686fc46faa62cebc6/frame/contracts/src/lib.rs#L392-L393